### PR TITLE
1565 - Add typeahead dropdown icon/button

### LINKED ISSUE #1565: Usability Typeahead widget: add select button
**Enhancement**

Typeahead widgets lack a visual clue that they are selection fields. When Typeahead is not used as an input helper for free text (e.g. input helper for mail recipients) but used in validation mode as a selection widget, the user wants a visual clue the field can be clicked. I suggest adding a button - exactly like in the Select widget - for this purpose. That might be configurable or an opinionated behavior (validating=show button, free text=no button).

Package Version: 7.0.3

**Code**

n/a

**Expected behavior:**

Show selection button like Select widget when Typeahead is configured as validating (or make it configurable)

**Actual behavior:**

No selection button is displayed, no visual cue.

### PR DESCRIPTION
**Pull request by KaneFreeman**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] For new widgets, `theme.variant()` is added to the root domnode
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:**
Add typeahead dropdown icon/button

Resolves #1565

### CODE REVIEW

File: src/typeahead/index.tsx [hunk 1]
```diff
@@ -418,7 +431,28 @@ export const Typeahead = factory(function Typeahead({
 								}}
 								valid={valid}
 							>
-								{{ label, leading }}
+								{{
+									label,
+									leading,
+									trailing: showDropDownButton && (
+										<button
```
Comment (agubler) [inline src/typeahead/index.tsx:438]: We couldn't use the dojo button here?

File: src/typeahead/index.tsx [hunk 2]
```diff
@@ -53,6 +55,8 @@ export interface TypeaheadProperties {
 	itemDisabled?: ListProperties['disabled'];
 	/** Flag to indicate if values other than those in the resource can be entered, defaults to true */
 	strict?: boolean;
+	/** Flag to indicate if drop down arrow should be shown in trailing section of text input, defaults to false */
+	showDropDownButton?: boolean;
```
Comment (agubler) [inline src/typeahead/index.tsx:59]: feels like a bit of a mouthful! I don't have a better idea other than we could potentially drop `show`

Comment (matt-gadd) [reply src/typeahead/index.tsx:59]: I prefer dijit's `hasDownArrow` if that helps.

Comment (agubler) [reply src/typeahead/index.tsx:59]: @matt-gadd That does help! Thanks, I'm sold.

**Review by agubler — CHANGES_REQUESTED** [review]
A couple of comments

File: src/typeahead/index.tsx (hunk 1 above)
Comment (KaneFreeman) [reply src/typeahead/index.tsx:439]: This is following the same pattern that the `Select` widget is using. If we want to change that pattern both should be changed together.

File: src/theme/dojo/typeahead.m.css [hunk 3]
```diff
@@ -5,3 +5,27 @@
...
+	line-height: var(--line-height-base);
+	padding: var(--spacing-regular);
+
+	cursor: pointer;
+	outline: none;
+
+	background: unset;
+
+	appearance: none;
+	-moz-appearance: none;
```
Comment (agubler) [inline src/theme/dojo/typeahead.m.css:25]: do we need these? doesn't the build generate these for us? cc @matt-gadd

**Review by agubler — APPROVED** [review]
One question but otherwise :+1:

File: src/theme/dojo/typeahead.m.css (hunk 3 above)
Comment (KaneFreeman) [reply src/theme/dojo/typeahead.m.css:25]: It does not appear to. I just tested removing them and they are not in the end product.

Several other widgets have them in their css too.

**Pull request merged**